src/main/java/org/openstreetmap/atlas/checks/validation/areas/ConcerningAngleBuildingCheck.java

```diff
@@ -0,0 +1,234 @@
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return !isFlagged(object.getOsmIdentifier()) && this.isBuildingOrPart(object)
+                && (object instanceof Area
+                        || (object instanceof Relation && ((Relation) object).isMultiPolygon()));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        markAsFlagged(object.getOsmIdentifier());
+        final Set<Polygon> buildingPolygons = this.getPolygons(object).collect(Collectors.toSet());
+        if (!buildingPolygons.isEmpty())
+        {
+            for (final Polygon polygon : buildingPolygons)
+            {
+                if (this.buildingAngleCountWithinValidRange(polygon)
+                        && this.hasConcerningAngles(polygon))
+                {
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(0, object.getOsmIdentifier())));
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * checks to make sure building node count fits within desired range
+     * 
+     * @param polygon
+     *            building being checked
+     * @return boolean ensuring that the node count fits within desired range
+     */
+    private boolean buildingAngleCountWithinValidRange(final Polygon polygon)
+    {
+        final List<Segment> polygonSegments = polygon.segments();
+        return polygonSegments.size() >= this.minAngleCount
+                && polygonSegments.size() <= this.maxAngleCount;
+    }
+
+    /**
+     * Get angle diff between 2 segments
+     * 
+     * @param segment1
+     *            a segment
+     * @param segment2
+     *            a connecting segment
+     * @return angle difference between headings of segments
+     */
+    private double getAngleDiff(final Segment segment1, final Segment segment2)
+    {
+        final Optional<Heading> segmentOneHeading = segment1.heading();
+        final Optional<Heading> segmentTwoHeading = segment2.heading();
+        if (segmentOneHeading.isPresent() && segmentTwoHeading.isPresent())
+        {
+            return segmentOneHeading.get().difference(segmentTwoHeading.get()).asDegrees();
+        }
+        return ANGLE_DEFAULT;
```

Comment:
I think we may run into a situation where the vertex of 2 segments contains 2 nodes --1 node belonging to the meeting segments and another stray node which would be a Segment of length 0. So there's actually 3 segments all connected. We could log something so it's clear why we're skipping this kind of polygon